torch/csrc/jit/passes/utils/memory_dag.cpp

```diff
@@ -189,6 +203,7 @@ void MemoryDAG::setWildcards(
   // For every element, if the cache contains `MemoryLocationFoo`, then we must
   // add `WildcardBar` to it.
   for (const std::unique_ptr<Element>& e : this->indexToElementMap_) {
+    e->cachedAllContainedMemoryLocations_.reset();
```

Comment:
Why do we need to do this ?

Reply:
we're mutating the graph, so we have to invalidate the caches (or update them, but that seems difficult to do efficiently)